cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -96,6 +102,12 @@ public void createIfNotExists(Id.DatasetInstance instance, String type,
     this.allowDatasetUncheckedUpgrade = conf.getBoolean(Constants.Dataset.DATASET_UNCHECKED_UPGRADE);
   }
 
+  @SuppressWarnings("unused")
+  @Inject(optional = true)
```

Comment:
By making the injection optional, we don't have to inject `AuditModule` into all unit tests.